l3kernel/l3keys.dtx

```diff
@@ -2858,40 +2861,47 @@
 %    \item The code to set up the correct selection approach
 %  \end{enumerate}
 %    \begin{macrocode}
-\cs_new_protected:Npn \@@_set:nnnnNn
+\cs_new_protected:Npn \@@_set:nnnnNn #1#2#3#4#5#6
   {
-    \exp_args:Nooo \@@_set:nnnnnnnNn
-      \l_@@_unused_clist
-      \l_@@_selective_clist
-      \l_@@_relative_tl
-  }
-\cs_new_protected:Npn \@@_set:nnnnnnnNn #1#2#3#4#5#6#7#8#9
-  {
-    \clist_clear:N \l_@@_unused_clist
-    \clist_set:Ne \l_@@_selective_clist { \tl_to_str:n {#5} }
-    \tl_set:Nn \l_@@_relative_tl {#7}
     \use:e
       {
         \exp_not:n
           {
-            #9
-            \@@_set:nn {#4} {#6}
+            \clist_clear:N \l_@@_unused_clist
+            \clist_set:Ne \l_@@_selective_clist { \tl_to_str:n {#2} }
+            \tl_set:Nn \l_@@_relative_tl {#4}
+            #6
+            \@@_set:nn {#1} {#3}
+            \clist_set_eq:NN #5 \l_@@_unused_clist
           }
         \@@_reset_bool:N \l_@@_only_known_bool
         \@@_reset_bool:N \l_@@_exclude_bool
         \@@_reset_bool:N \l_@@_selective_bool
+        \@@_reset_var:N \l_@@_unused_clist
+        \@@_reset_var:N \l_@@_selective_clist
+        \@@_reset_var:N \l_@@_relative_tl
+        \@@_reset_var:N \l_@@_inherit_str
+        \@@_reset_var:N \l_keys_choice_tl
+        \@@_reset_var:N \l_keys_key_tl
+        \@@_reset_var:N \l_keys_key_str
+        \@@_reset_var:N \l_keys_path_tl
+        \@@_reset_var:N \l_keys_path_str
+        \@@_reset_var:N \l_keys_value_tl
+        \int_set:Nn \l_keys_choice_int { \int_use:N \l_keys_choice_int }
       }
-    \clist_set_eq:NN #8 \l_@@_unused_clist
-    \__kernel_tl_set:Nx \l_@@_unused_clist { \exp_not:n {#1} }
-    \__kernel_tl_set:Nx \l_@@_selective_clist {#2}
-    \__kernel_tl_set:Nx \l_@@_relative_tl { \exp_not:n {#3} }
   }
 \cs_new:Npn \@@_reset_bool:N #1
   {
     \exp_not:c
       { bool_set_ \bool_if:NTF #1 { true } { false } :N }
         \exp_not:N #1
   }
+\cs_new:Npn \@@_reset_var:N #1
```

Comment:
`\begin{macro}` not updated.

Also `\begin{macro}{\@@_set:nnnnnnnNn}` and an `\end{macro}` need to be removed.

Reply:
`\@@_reset_var:N` was already in a `\begin{macro}` but never actually defined. Removed the obsolete `macro` environment.

Reply:
You're right. It was added by efe0c495 (Re-factor selective key setting code, 2024-05-08).